src/librustc/mir/visit.rs

```diff
@@ -847,6 +847,7 @@ macro_rules! make_mir_visitor {
             fn visit_location(&mut self, body: & $($mutability)? Body<'tcx>, location: Location) {
                 let basic_block = & $($mutability)? body[location.block];
                 if basic_block.statements.len() == location.statement_index {
+                    // TODO(nashenas88) how to ensure we clear the cache only in the mutable case...
```

Comment:
@oli-obk this one I'm not sure on a proper path forward

Reply:
you can do `if stringify!($mutability) == "mut"` :laughing: it'll get compiled out by llvm.

alternatively you can create a macro that emits either `body.basic_block(location.block)` or `body.basic_block_mut(location.block)` depending on the value of `$mutability` (that's probably a cleaner solution

Reply:
Unfortunately rustc couldn't understand whether that code would be compiled or not. I had to make a new macro rule and call that instead. Should be up shortly (once compilation is done)

Reply:
Specifically the borrow checker was not happy with an `&mut` reference from a type that was only `&`